### PR TITLE
Improve Documentation for Ethereum Solidity Contracts

### PR DESCRIPTION
**Pull request by danintel**
Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: examples/common/python/connectors/TestingContracts.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
-# Test Process
+# Solidity Connector Test Process
 
-1. Installing Python 3.6.8
+1. Install Python 3.6.8 if not currently installed.
```
[the remark this answers is not in the document]
Comment (danintel) [reply examples/common/python/connectors/TestingContracts.md]: @manojgop Only Step 1 is performed by the Docker-based build.
web3.py is not installed (step 2)
The solc compiler is not installed (step 3)
SOLC_BINARY is not set (step 4).

For example:

```
$ sudo docker exec -it tcf bash
root@f48d605bea9e:/project/TrustedComputeFramework/tools/build#  source _dev/bin/activate
(_dev) root@f48d605bea9e:/project/TrustedComputeFramework/tools/build# ls -a ~/.py-solc
ls: cannot access '/root/.py-solc': No such file or directory
(_dev) root@f48d605bea9e:/project/TrustedComputeFramework/tools/build# echo $SOLC_BINARY
```

[the remark this answers is not in the document]
Comment (Ram-srini) [reply examples/common/python/connectors/TestingContracts.md]: @danintel For docker based build these steps are not required. Because we are already installed python, web3 and py-solc in docker images. These are part of image.
(_dev) root@0840863d8f74:/project/TrustedComputeFramework/tools/build# pip list web3 |grep web3
web3                5.0.2
(_dev) root@0840863d8f74:/project/TrustedComputeFramework/tools/build# echo $SOLC_BINARY
/root/.py-solc/solc-v0.4.25/bin/solc
(_dev) root@0840863d8f74:/project/TrustedComputeFramework/tools/build# ls $SOLC_BINARY
/root/.py-solc/solc-v0.4.25/bin/solc

[the remark this answers is not in the document]
Comment (danintel) [reply examples/common/python/connectors/TestingContracts.md]: These steps are required until the Docker scripts are fixed to support proxies.
They work only when ran on a machine directly connected to Internet, but not on a machine inside a corporate proxy.  I created this JIRA bug for this problem:
https://jira.devtools.intel.com/browse/TCFH-413

[the remark this answers is not in the document]
Comment (danintel) [reply examples/common/python/connectors/TestingContracts.md]: From JIRA TCFH-413:
I found the problem. The `BUILD.md` instructions are missing:
`docker-compose up --build`
Otherwise the `docker/Dockerfile.tcf-dev` file is not ran.
This occurs if an old Docker image still exists.
I am fixing steps in `docker/Dockerfile.tcf-dev` as the OpenSSL build fails–OpenSSL is getting compiler errors from missing definitions while building.
This could be because `Dockerfile.tcf-dev` uses openssl-1.1.1b instead of openssl-1.1.0h, which works and is specified in PREREQUISITES.md.
Or the problem may be `-j$THREADS` (8 threads) is causing a failure?

There are several differences between the way Dockerfile.tcf-dev builds and the instructiuons in PRERREQUISITES.md, so it may take awhile to find out where the problem is located.

There could also be a proxy problem while building.

Basically, there are several problems with Dockefile.tcf-dev, it appears.

[the remark this answers is not in the document]
Comment (Ram-srini) [reply examples/common/python/connectors/TestingContracts.md]: If you changed anything in Dockerfile then we should re-create the image and start the container using command 'docker-compose up --build'. BTW we are using NUCs which are also behind proxy, we are facing any proxy related issues.
